shell/machine-config/azure.vue

```diff
@@ -156,24 +163,32 @@ export default {
         this.value.tenantId = tenantId;
       }
 
-      this.locationOptions = await this.$store.dispatch('management/request', {
-        url:    addParam('/meta/aksLocations', 'cloudCredentialId', this.credentialId),
-        method: 'GET',
-      });
+      if (this.loadedCredentialIdFor !== this.credentialId) {
+        this.locationOptions = await this.$store.dispatch('management/request', {
+          url:    addParam('/meta/aksLocations', 'cloudCredentialId', this.credentialId),
+          method: 'GET',
+        });
 
-      if (this.mode === _CREATE) {
-        this.value.location = DEFAULT_REGION;
-
-      // when you edit an Azure cluster and add a new machine pool (edit)
-      // the location field doesn't come populated which causes the vmSizes request
-      // to return 200 but with a null response (also a bunch of other fields are undefined...)
-      // so let's prefill them with the defaults
-      } else if (this.mode === _EDIT && !this.value?.location) {
-        for (const key in this.defaultConfig) {
-          if (this.value[key] === undefined) {
-            this.$set(this.value, key, this.defaultConfig[key]);
-          }
-        }
+        this.loadedCredentialIdFor = this.credentialId;
+      }
+
+      // if (this.mode === _CREATE) {
+      //   this.value.location = DEFAULT_REGION;
+
+      // // when you edit an Azure cluster and add a new machine pool (edit)
+      // // the location field doesn't come populated which causes the vmSizes request
+      // // to return 200 but with a null response (also a bunch of other fields are undefined...)
+      // // so let's prefill them with the defaults
+      // } else if (this.mode === _EDIT && !this.value?.location) {
+      //   for (const key in this.defaultConfig) {
+      //     if (this.value[key] === undefined) {
+      //       this.$set(this.value, key, this.defaultConfig[key]);
+      //     }
+      //   }
+      // }
```

Comment:
I think we still need this block: when adding a machine pool on edit defaults are not set and incorrect errors appear:
![Screen Shot 2023-12-14 at 11 57 09 AM](https://github.com/rancher/dashboard/assets/42977925/097bcf05-871d-42c1-a6be-5e904231bfa8)

From my testing, the location is still set correctly with this code back in place.